src/dbnode/storage/series/series.go

```diff
@@ -373,89 +379,14 @@ func (s *dbSeries) addBlockWithLock(b block.DatabaseBlock) {
-	}
-
-	s.Lock()
-	s.loadWithLock(false, blocksToLoad, blockStates)
-	s.Unlock()
-	return LoadResult{}, nil
-}
-
-func (s *dbSeries) bootstrap(
-	bootstrappedBlocks block.DatabaseSeriesBlocks,
-	blockStates BootstrappedBlockStateSnapshot,
-) (BootstrapResult, error) {
+func (s *dbSeries) LoadBlock(
+	block block.DatabaseBlock,
+	writeType WriteType,
+) error {
 	s.Lock()
-	defer func() {
-		s.bs = bootstrapped
-		s.Unlock()
-	}()
-
-	var result BootstrapResult
-	if s.bs == bootstrapped {
-		return result, errSeriesAlreadyBootstrapped
-	}
-
-	if bootstrappedBlocks == nil {
-		return result, nil
-	}
-
-	s.loadWithLock(true, bootstrappedBlocks, blockStates)
-	result.NumBlocksMovedToBuffer += int64(bootstrappedBlocks.Len())
-
-	return result, nil
-}
-
-func (s *dbSeries) loadWithLock(
-	isBootstrap bool,
-	blocksToLoad block.DatabaseSeriesBlocks,
-	blockStates BootstrappedBlockStateSnapshot,
-) {
-	for _, block := range blocksToLoad.AllBlocks() {
-		if !isBootstrap {
-			// The data being loaded is not part of the bootstrap process then it needs to be
-			// loaded as a cold write because the load could be happening concurrently with
-			// other processes like the flush (as opposed to bootstrap which cannot happen
-			// concurrently with a flush) and there is no way to know if this series/block
-			// combination has been warm flushed or not yet since updating the shard block state
-			// doesn't happen until the entire flush completes.
-			//
-			// As a result the only safe operation is to load the block as a cold write which
-			// ensures that the data will eventually be flushed and merged with the existing data
-			// on disk in the two scenarios where the Load() API is used (cold writes and repairs).
-			s.buffer.Load(block, ColdWrite)
-			continue
-		}
-
-		blStartNano := xtime.ToUnixNano(block.StartTime())
-		blState := blockStates.Snapshot[blStartNano]
-		if !blState.WarmRetrievable {
-			// If the block being bootstrapped has never been warm flushed before then the block
-			// can be loaded into the buffer as a WarmWrite because a subsequent warm flush will
-			// ensure that it gets persisted to disk.
-			//
-			// If the ColdWrites feature is disabled then this branch should always be followed.
-			s.buffer.Load(block, WarmWrite)
-		} else {
-			// If the block being bootstrapped has been warm flushed before then the block should
-			// be loaded into the buffer as a ColdWrite so that a subsequent cold flush will ensure
-			// that it gets persisted to disk.
-			//
-			// This branch can be executed in the situation that a cold write was received for a block
-			// that had already been flushed to disk. Before the cold write could be persisted to disk
-			// via a cold flush, the node crashed and began bootsrapping itself. The cold write would be
-			// read out of the commitlog and would eventually be loaded into the buffer via this branch.
-			s.buffer.Load(block, ColdWrite)
-		}
-	}
+	defer s.Unlock()
```

Comment:
nit: can probably remove the defer here?

Reply:
Going to keep as is for expedited merging.